load_tests/load_test.py

```diff
@@ -110,53 +147,102 @@ def run_ecs_tests():
+                os.environ['DESTINATION'] = 's3'
+            processes.add(subprocess.Popen(['go', 'run', './load_tests/validation/validate.go', input_record, log_delay]))
+        
+        # Wait until all subprocesses for validation completed
+        for p in processes:
+            p.wait()
 
-    # Wait until task stops and start validation
-    for throughput in THROUGHPUT_LIST:
-        waiter.wait(
-            cluster=ecs_cluster_name,
-            tasks=[
-                names[f'{OUTPUT_PLUGIN}_{throughput}_task_arn'],
-            ],
-            WaiterConfig={
-                'MaxAttempts': 600
-            }
-        )
-        response = client.describe_tasks(
-            cluster=ecs_cluster_name,
-            tasks=[
-                names[f'{OUTPUT_PLUGIN}_{throughput}_task_arn'],
-            ]
+# Returns s3 arn
+def publish_fluent_config_s3(input_logger):
+    bucket_name = os.environ['S3_BUCKET_NAME']
+    s3 = boto3.client('s3')
+    s3.upload_file(
+        input_logger["fluent_config_file_path"],
+        bucket_name,
+        f'{OUTPUT_PLUGIN}-test/{PLATFORM}/fluent-{input_logger["name"]}.conf',
+    )
+    return f'arn:aws:s3:::{bucket_name}/{OUTPUT_PLUGIN}-test/{PLATFORM}/fluent-{input_logger["name"]}.conf'
+
+# The following method is used to clear data between
+# testing batches
+def delete_testing_data():
+    # All testing data related to the plugin option will be deleted
+    if OUTPUT_PLUGIN == 'cloudwatch':
+        # Delete associated cloudwatch log streams
+        client = boto3.client('logs')
+        response = client.describe_log_streams(
+            logGroupName=os.environ['CW_LOG_GROUP_NAME']
         )
-        check_app_exit_code(response)
-        input_record = calculate_total_input_number(throughput)
-        log_delay = get_log_delay(response)
-        set_buffer(response)
-        # Validate logs
-        if OUTPUT_PLUGIN == 'cloudwatch':
-            os.environ['LOG_PREFIX'] = throughput
-            os.environ['DESTINATION'] = 'cloudwatch'
-        else:
-            os.environ['LOG_PREFIX'] = f'{OUTPUT_PLUGIN}-test/ecs/{throughput}/'
-            os.environ['DESTINATION'] = 's3'
-        processes.add(subprocess.Popen(['go', 'run', './load_tests/validation/validate.go', input_record, log_delay]))
-    
-    # Wait until all subprocesses for validation completed
-    for p in processes:
-        p.wait()
+        for stream in response["logStreams"]:
+            client.delete_log_stream(
+                logGroupName=os.environ['CW_LOG_GROUP_NAME'],
+                logStreamName=stream["logStreamName"]
+            )
+    else:
+        # Delete associated s3 bucket objects
+        s3 = boto3.resource('s3')
+        bucket = s3.Bucket(os.environ['S3_BUCKET_NAME'])
+        firehose_objects = bucket.objects.filter(Prefix=f'{OUTPUT_PLUGIN}-test/{PLATFORM}/')
+        firehose_objects.delete()
+        return
 
 def delete_testing_resources():
```

Comment:
I am thinking about if we should keep the logic here as `delete_testing_data` as well:
* cloudwatch codebuild project deletes cw group & ecs cluster
* s3, kinesis, firehose plugins delete each folder generated by the output plugin

Reply:
Discussed separately, Zhonghui explained how the system works and we think that the delete stack will also delete the cloudwatch log group and get rid of the data.

Reply:
What created the log group? if Fluent Bit created it with `auto_create_group On` (which I see in this PR) then FB created it and CFN delete stack will not delete it. CFN delete stack only deletes things which the stack explicitly created... we should use that, create all log groups in CFN and then set `auto_create_group Off`. This is the best practice actually, and its what CDK does for the awslogs driver for example IIRC, it doesn't use the auto create group option on the log group and instead provisions the log group in the stack, which allows you to delete it later when you delete the stack.

Reply:
@PettitWesley we created the log group here: https://github.com/aws/aws-for-fluent-bit/blob/mainline/load_tests/create_testing_resources/app.py#L16. We set the generated log group in task definition and yeah, I guess we should set `auto_create_group Off`

Reply:
Still need to do this. Will change after the Improvement PR is merged in: https://github.com/fala-aws/aws-for-fluent-bit/pull/2